ngi_reports/stockholm/project_summary.py

```diff
@@ -161,64 +163,68 @@ def __init__(self, config, LOG, working_dir, **kwargs):
+        sample_qstat_keys['HiSeqX'] = {'qval':'% >= Q30bases', 'bases':'PF Clusters'}
+
+        ## Collect required information for all flowcell run for the project
         for fc in self.flowcell_info.values():
             fc_name = fc['name']
-            fc_obj = fcon.get_entry(fc['run_name'])
-            fc_runp = fc_obj.get('RunParameters',{})
+            fc_obj = xcon.get_entry(fc['run_name']) if fc['type'] == 'HiSeqX' else fcon.get_entry(fc['run_name'])
+            fc_runp = fc_obj.get('RunParameters',{}).get('Setup',{}) if fc['type'] == 'HiSeqX' else fc_obj.get('RunParameters',{})
             fc_illumina = fc_obj.get('illumina',{})
-            fc_run_summary = fc_illumina.get('run_summary',{})
-            self.flowcell_info[fc_name]['lanes'] = {}
-            
+            fc_lane_summary = fc_obj.get('lims_data', '').get('run_summary', {})
+            self.flowcell_info[fc_name]['lanes'] = OrderedDict()
+
             ## Get sequecing method for the flowcell
             seq_template = "{}) Clustering was done by '{}' and samples were sequenced on {} ({}) with a {} setup using '{}' "\
                            "chemistry. The Bcl to FastQ conversion was performed using {} from the CASAVA software suite. The "\
                            "quality scale used is Sanger / phred33 / Illumina 1.8+."
             run_setup = fc_obj.get("run_setup")
             fc_chem = fc_runp.get('ReagentKitVersion', fc_runp.get('Sbs'))
-            seq_plat = ["HiSeq2500","MiSeq"]["MCSVersion" in fc_runp.keys()]
+            seq_plat = fc['type']
             clus_meth = ["cBot","onboard clustering"][seq_plat == "MiSeq" or fc_runp.get("ClusteringChoice","") == "OnBoardClustering"]
-            casava = fc_obj.get('DemultiplexConfig',{}).values()[0].get('Software',{}).get('Version')
+            try:
+                casava = fc_obj.get('DemultiplexConfig',{}).values()[0].get('Software',{}).get('Version')
+            except:
+                casava = None
             if seq_plat == "MiSeq":
                 seq_software = "MSC {}/RTA {}".format(fc_runp.get("MCSVersion"),fc_runp.get("RTAVersion"))
-                self.proj_has_miseq = True
             else:
                 seq_software = "{} {}/RTA {}".format(fc_runp.get("ApplicationName"),fc_runp.get("ApplicationVersion"),fc_runp.get("RTAVersion"))
-                self.proj_has_hiseq = True
             tmp_method = seq_template.format("SECTION", clus_meth, seq_plat, seq_software, run_setup, fc_chem, casava)
-            
+
             ## to make sure the sequencing methods are unique
-            if tmp_method not in seq_methods.keys():
-                seq_methods[tmp_method] = alphabets[len(seq_methods.keys())]
-            self.flowcell_info[fc_name]['seq_meth'] = seq_methods[tmp_method]
-            
+            if tmp_method not in self.seq_methods.keys():
+                self.seq_methods[tmp_method] = alphabets[len(self.seq_methods.keys())]
```

Comment:
Yes, `self.seq_methods` will be an empty dict for the very first time and `len(self.seq_methods.keys())` will give a `0` and works fine for the list `alphabets`

The same for further iteration, as I calculate length before assigning it  :+1: